src/test/java/org/scijava/io/handle/FileHandleTest.java

```diff
@@ -108,4 +110,30 @@ public void testNotCreatedByClose() throws IOException {
 		handle.close();
 		assertFalse(nonExistentFile.exists());
 	}
+
+	@Test
+	public void testNotCreatedByRead() throws IOException {
+		final Context ctx = new Context();
+		final DataHandleService dhs = ctx.service(DataHandleService.class);
+
+		final File nonExistentFile = //
+			File.createTempFile("FileHandleTest", "none xistent file");
+		final URI uri = nonExistentFile.toURI();
+		assertTrue(nonExistentFile.delete());
+		assertFalse(nonExistentFile.exists());
+
+		final FileLocation loc = new FileLocation(uri);
+		final DataHandle<?> handle = dhs.create(loc);
```

Comment:
We should use try-with-resources here, no?
```
try (final DataHandle<?> handle = dhs.create(loc)) {
  ...
}
```
So that it definitely gets closed at the end, regardless of the outcome of the test